mart/attack/composer.py

```diff
@@ -7,12 +7,37 @@
 from __future__ import annotations
 
 import abc
+from collections import OrderedDict
 from typing import Any, Iterable
 
 import torch
 
 
-class Composer(abc.ABC):
+class Function(torch.nn.Module):
+    def __init__(self, *args, order=0, **kwargs) -> None:
+        """A stackable function for Composer.
+
+        Args:
+            order (int, optional): The priority number. A smaller number makes a function run earlier than others in a sequence. Defaults to 0.
+        """
+        super().__init__(*args, **kwargs)
+        self.order = order
+
+    @abc.abstractmethod
+    def forward(self, perturbation, input, target) -> None:
+        """Returns the modified perturbation, modified input and target, so we can chain Functions
+        in a Composer."""
+        pass
+
+
+class Composer:
+    def __init__(self, functions: dict[str, Function]) -> None:
```

Comment:
I would just do **names_and_functions, so you don't need to pass a dict. That is,

```python
Composer(mask=Mask(order=0), additive=Additive(order=1))
```

Reply:
Yes, but leaving `functions` here makes it much easier to do modality_dispatch later.